students/EricAdams/session06/test_mailroom.py

```diff
@@ -0,0 +1,221 @@
+            is used as input data.
+            The value that the user would input is simulated
+            by making a one entry into the file per user input.
+            Example: for a user input of 'list', enter 'list'
+            on the first line, (no quotes).
+            Enter 'menu' on the second line for the second user
+            input, (this also forces a return).
+    Modified from a solution to the 'input' issue on stackoverflow.
+    Seems to work.
+    '''
+    orig_stdin = sys.stdin
+    with open("test_send_thank_you_list_inputs.txt") as sys.stdin:
+        result = mailroom.send_thank_you()
+    sys.stdin = orig_stdin
+    test_print_donors(capsys)
+    assert result is None
+
+
+def test_send_thank_you_donor_existing_menu():
+    ''' Simulate a user input of an existing donor.
+    After the expected result simulate a user input of 'menu'.
+    This will cause the function to return.
+    :Data file = test_send_thank_you_donor_existing_menu.txt in
+                 current working directory
+    '''
+    orig_stdin = sys.stdin
+    with open("test_send_thank_you_donor_existing_menu.txt") as sys.stdin:
+        result = mailroom.send_thank_you()
+    sys.stdin = orig_stdin
+    assert result is None
+
+
+def test_send_thank_you_donor_existing_amount_no_write():
+    '''Simulate a user input of an existing donor.
+    Simulate a user input of a donation amount.
+    :Data file = test_send_thank_you_donor_existing_amount_no_write.txt
+                 in current working directory
+    '''
+    orig_stdin = sys.stdin
+    with open("test_send_thank_you_donor_existing_amount_no_write.txt") \
+            as sys.stdin:
+        result = mailroom.send_thank_you()
+    sys.stdin = orig_stdin
+    with open("test_send_thank_you_donor_existing_amount_no_write.txt") \
+            as f:
+        donor = f.readline()
+        donor = donor.strip()
+        amount = f.readline()
+    assert mailroom.donor_db[donor][-1] == float(amount)
+    assert result is None
+
+
+def test_send_thank_you_new_donor_amount_no_write():
+    '''Simulate a user input of a new donor.
+    Simulate a user input of a donation amount.
+    :Data file = test_send_thank_you_new_donor_amount_no_write.txt
+                 First line of file = new donor name
+                 Second line of file = donation amount (float)
+                 Third line of file = n, (no quotes)
+                 File is in current working directory
+    '''
+    orig_stdin = sys.stdin
+    with open("test_send_thank_you_new_donor_amount_no_write.txt") \
+            as sys.stdin:
+        result = mailroom.send_thank_you()
+    sys.stdin = orig_stdin
+    with open("test_send_thank_you_donor_existing_amount_no_write.txt") \
+            as f:
+        donor = f.readline()
+        donor = donor.strip()
+        amount = f.readline()
+    assert mailroom.donor_db[donor][-1] == float(amount)
+    assert result is None
+
+
+def test_send_thank_you_print_letter_no_write(capsys):
```

Comment:
does this need to be tested?

in a way, you are testing that print() works :-)

If you have tests that test that the letter is correctly generated, then maybe no need to test that it gets printed to the console correctly.